server/api/controllers/reports.js

```diff
@@ -0,0 +1,37 @@
+import { ArticleReports, Article } from '../../models';
+
+/**
+ *
+ * @function reportArticle
+ * @param {*} req any
+ * @param {*} res  any
+ * @returns {Object} JSON object
+ */
+export const reportArticle = async (req, res) => {
+  try {
+    const userId = req.user.id;
+    const articleId = req.params.id;
+    const newReport = {
+      userId,
+      articleId
+    };
+    const article = await Article.findById(articleId);
+    if (!article) {
+      throw new Error('Article is not found');
+    }
+
+    const articleReport = await ArticleReports.findOne({ where: { userId, articleId }, defaults: { newReport } });
+    if (articleReport) {
+      throw new Error(`Hello ${req.user.firstname} You've already reported this article`);
```

Comment:
@Princesso Consider returning these errors to the client. Using the `new Error` syntax will only kill the process and throw the error stack on the console. That's not the behaviour we want on this.

Reply:
The throw error method, returns a custom message to the user when a message is passed into it. I have tested this and it works. It was actually born out of the fact that I had nested if statements and while refactoring, one of the things I try to do while refactoring is to remove all nested if statements. we could look at it together.